### PR TITLE
v2

### PR DESCRIPTION
**Pull request by DoctorMcKay**
- Now requires node 12 or later
- Added support for BigInts
- Removed dependency on cuint

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,244 +1,370 @@
...
 		}
 
-		if (typeChar == 'c') {
-			this.instance |= SteamID.ChatInstanceFlags.Clan;
-			this.type = SteamID.Type.CHAT;
-		} else if (typeChar == 'L') {
-			this.instance |= SteamID.ChatInstanceFlags.Lobby;
-			this.type = SteamID.Type.CHAT;
+		let matches;
+		if (typeof input == 'bigint' || (typeof input == 'string' && input.match(/^\d+$/))) {
```
Comment (Revadike) [inline index.js:115]: Isnan or isfinite instead of regex better maybe?

File: index.js [hunk 2]
```diff
@@ -1,244 +1,370 @@
...
+			this.universe = Number(num >> 56n);
+		} else if ((matches = input.match(/^STEAM_([0-5]):([0-1]):([0-9]+)$/))) {
+			// Steam2 ID
+			let [_, universe, mod, accountid] = matches;
+
+			this.universe = parseInt(universe, 10) || SteamID.Universe.PUBLIC; // If it's 0, turn it into 1 for public
+			this.type = SteamID.Type.INDIVIDUAL;
+			this.instance = SteamID.Instance.DESKTOP;
+			this.accountid = (parseInt(accountid, 10) * 2) + parseInt(mod, 10);
+		} else if ((matches = input.match(/^\[([a-zA-Z]):([0-5]):([0-9]+)(:[0-9]+)?]$/))) {
```
Comment (Revadike) [inline index.js:130]: `=` ? I'm so confused by this

**Review by Revadike — APPROVED** [review]
Looks great

File: index.js (hunk 2 above)
Comment (DoctorMcKay) [reply index.js:130]: It's basically this, but all on one line:
```js
matches = input.match(...);
if (matches) {
```

File: index.js (hunk 1 above)
Comment (DoctorMcKay) [reply index.js:115]: Negative numbers would pass both of those.

**Pull request merged**